app/src/main/java/com/neura/sampleapplication/NeuraManager.java

```diff
@@ -46,10 +63,160 @@ public NeuraApiClient getClient() {
      * the settings). Fyi this only means that the sensors are disabled, not when there's no wifi available fe.
      */
     public void initNeuraConnection(Context context) {
+        if (!isMinVersion()) {
+            return;
+        }
+
         Builder builder = new Builder(context);
         mNeuraApiClient = builder.build();
         mNeuraApiClient.setAppUid(context.getResources().getString(R.string.app_uid));
         mNeuraApiClient.setAppSecret(context.getResources().getString(R.string.app_secret));
         mNeuraApiClient.connect();
     }
-}
+
+    //create a call back to handle authentication stages.
+    private static AnonymousAuthenticationStateListener silentStateListener = new AnonymousAuthenticationStateListener() {
+        @Override
+        public void onStateChanged(AuthenticationState state) {
+
+            switch (state) {
+                case AccessTokenRequested:
+                    break;
+                case AuthenticatedAnonymously:
+                    // successful authentication
+                    NeuraManager.getInstance().getClient().unregisterAuthStateListener();
+
+//                    NeuraManager.getInstance().getClient().getUserDetails(new UserDetailsCallbacks() {
+//                        @Override
+//                        public void onSuccess(UserDetails userDetails) {
+//                            if (getActivity() != null) {
+//                                userDetails.getData().getNeuraId();
+//                                NeuraManager.getInstance().getClient().getUserAccessToken();
+//                            }
+//                        }
+//
+//                        @Override
+//                        public void onFailure(Bundle resultData, int errorCode) {
+//
+//                        }
+//                    });
+                    break;
+                case NotAuthenticated:
+                case FailedReceivingAccessToken:
+                    // Authentication failed indefinitely. a good opportunity to retry the authentication flow
+                    NeuraManager.getInstance().getClient().unregisterAuthStateListener();
+                    break;
+                default:
+            }
+        }
+    };
+
+    public static void authenticateAnonymously(final FragmentMain context) {
```

Comment:
@LeonidJack Is there any other way than passing the fragment context to this function?
it narrows the options to use it only in UI based systems

Reply:
The fragment context is there purely to replicate the response to successful authentication in phone authentication. If the developer doesn't want it to be connected to the UI, they can remove the parameter and any lines of code that used this context. The authentication can happen without it.